Code/Framework/AzCore/AzCore/Jobs/Internal/JobTypeEraser.h

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Jobs/JobDescriptor.h>
+#include <AzCore/std/containers/fixed_vector.h>
+#include <AzCore/std/typetraits/is_assignable.h>
+#include <AzCore/std/typetraits/is_destructible.h>
+#include <AzCore/std/parallel/atomic.h>
+#include <AzCore/Memory/SystemAllocator.h>
+
+namespace AZ::Internal
+{
+    using JobInvoke_t = void (*)(void* lambda);
+    using JobRelocate_t = void (*)(void* dst, void* src);
+    using JobDestroy_t = void (*)(void* obj);
+
+    class CompiledJobGraph;
+
+    // Lambdas are opaque types and we cannot extract any member function pointers. In order to store lambdas in a
```

Comment:
Well it is possible to get the `operator()` member of a lambda since it is still a real class that just doesn't have a name.
We extract out the function signature for a lambda in our [function_traits](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/typetraits/function_traits.h#L107-L110) implementation by examining the member function pointer to the `operator()` function.

Reply:
There are complications here though. First, the size of a member function pointer isn't guaranteed to be 16 bytes on every platform (it certainly isn't 8 on every platform). I intentionally avoided parsing the template signature type here actually to improve compiler throughput.